### ISSUE TITLE
error while generating  random signal and write it to the buffer 

### ISSUE THREAD

**Issue by mfibmm** [fieldtrip/fieldtrip#2250]
**Describe the bug**
i am trying to use this example to genrate buffer data
The easiest way to try out the [ft_realtime_powerestimate](https://www.fieldtriptoolbox.org/reference/realtime/example/ft_realtime_powerestimate) example is by starting two MATLAB sessions. In the first session you create some random signal and write it to the buffer by means of [ft_realtime_signalproxy](https://www.fieldtriptoolbox.org/reference/realtime/example/ft_realtime_signalproxy):

cfg                = [];
cfg.channel        = 1:10;                         % list with channel "names"
cfg.blocksize      = 1;                            % seconds
cfg.fsample        = 250;                          % sampling frequency, Hz
cfg.lpfilter       = 'yes';                        % apply a low-pass filter
cfg.lpfreq         = 20;                           % filter frequency, Hz
cfg.target.dataset = 'buffer://localhost:1972';    % where to write the data
ft_realtime_signalproxy(cfg)
In the second MATLAB session you start the [ft_realtime_powerestimate](https://www.fieldtriptoolbox.org/reference/realtime/example/ft_realtime_powerestimate) and point it to the buffer:

cfg                = [];
cfg.blocksize      = 1;                            % seconds
cfg.foilim         = [0 30];                       % frequency-of-interest limits, Hz
cfg.dataset        = 'buffer://localhost:1972';    % where to read the data from
ft_realtime_powerestimate(cfg)
ft_realtime_signalproxy(cfg)
**To Reproduce**

**Expected behavior**
I was expecting to plot the buffer data, but the i got the error attached, this problem is not find in Fieldtrip 20180602 , it means inly this problem in the newer fieldtrip version.

**Screenshots**

**Environment and versions (please complete the following information):**
 - Operating System: [e.g. Windows 10]
 - MATLAB version [e.g. R2022b, type `ver`]
 - FieldTrip version [e.g. 2020202, or "most recent master branch from github", type `ft_version`]

**Additional context**

Error using cell/ismember
Input A of class cell and input B of class double must be cell arrays of character vectors, unless one
is a character vector.

Error in ft_senstype (line 359)
    if     (mean(ismember(ft_senslabel('ant128'),         sens.label)) > 0.8)

Error in ft_chantype (line 141)
elseif ft_senstype(input, 'unknown')

Error in ft_write_data (line 110)
  hdr.chantype = ft_chantype(hdr);

Error in ft_realtime_signalproxy (line 178)
    ft_write_data(cfg.target.datafile, dat, 'header', hdr, 'dataformat', cfg.target.dataformat,
    'append', false);

**Comment by mfibmm** [comment]
the problem was solved when I  replaced the file **ft_write_data** from Fieldtrip 20220202  by the one in 20180602

**Comment by schoffelen** [comment]
Perhaps you could investigate the commit history of this file for the period between 20180602 and 20220202 to diagnose this further

**Comment by schoffelen** [comment]
if you change cfg.channel from a numeric vector into a cell-array, such as cfg.channel = {'a';'b';'c';'d';'e';'f';'g';'h';'i';'j'}
then it works for me, also for the current (i.e. 2023) version of the code

can you please updated the documentation? Thanks.

**Comment by mfibmm** [comment]
yes the bug is solved when a cell-array, such as cfg.channel = {'a';'b';'c';'d';'e';'f';'g';'h';'i';'j'} is used as set of channels

**Comment by mfibmm** [comment]
> if you change cfg.channel from a numeric vector into a cell-array, such as cfg.channel = {'a';'b';'c';'d';'e';'f';'g';'h';'i';'j'} then it works for me, also for the current (i.e. 2023) version of the code
> 
> can you please updated the documentation? Thanks.

thanks the bug is solved when a cell-array, such as cfg.channel = {'a';'b';'c';'d';'e';'f';'g';'h';'i';'j'} is used as set of channels

**Comment by schoffelen** [comment]
Yes I know, can you please suggest a fix to the documentation ?

**Comment by mfibmm** [comment]
I do not know hiw to do this, could you please help me to handle this?

Accept all My Regards

​​Mohamed Issa
Postdoctoral Researcher at BCBL
www.bcbl.eu

Legal disclaimer/Aviso legal/Lege-oharra: www.bcbl.eu/legal-disclaimer
We are working flexibly at BCBL. I am sending this message at a time that suits me. Please feel comfortable knowing that I do not expect you to read, respond to or action it outside of regular working hours.
________________________________

**Comment by mfibmm** [comment]
Bug fixed

**Comment by schoffelen** [comment]
There is nothing fixed at the moment.

**Comment by schoffelen** [comment]
https://github.com/fieldtrip/website/commit/0917b21a705d7d133dbacef8e803012c5b3e272f

Now it is fixed
